src/lockscreen.js

```diff
@@ -4,12 +4,17 @@
 const {Gio, GLib, GObject, Gtk} = imports.gi;
 
 const {Playbin} = imports.playbin;
+const {Toolbox} = imports.toolbox;
 const SoundServer = imports.soundServer;
 
 var Lockscreen = GObject.registerClass({
     GTypeName: 'Lockscreen',
     CssName: 'lockscreen',
     Properties: {
+        toolbox: GObject.ParamSpec.object(
+            'toolbox', 'Toolbox', '',
```

Comment:
is accessing the `target-app-id` from the lockscreen the goal here? If so, we should just add a `target-app-id` property and avoid all that unnecessary toolbox refactoring.

Note that:
1.  the window already carries that property https://github.com/endlessm/hack-toolbox-app/blob/master/src/window.js#L8
2.  there is already logic in place to pass that property to the lockscreen, e.g. https://github.com/endlessm/hack-toolbox-app/blob/master/src/framework/controlPanel.js#L46 (same thing happens for all toolboxes).

Reply:
Thanks. What you propose is better.